src/chat/bot.py

```
@@ -0,0 +1,78 @@
+from typing import final
+
+from google.genai.types import HarmBlockThreshold, HarmCategory
+from pydantic_ai import Agent
+from pydantic_ai.messages import UserPromptPart
+from pydantic_ai.models.google import GoogleModel, GoogleModelSettings
+from pydantic_ai.providers.google import GoogleProvider
+from pydantic_ai.toolsets import AbstractToolset
+
+from chat.memory import retrieve_conversation, set_conversation
+from chat.memory import retrieve_conversation, set_conversation
```

Ruff (0.12.2)

Redefinition of unused `retrieve_conversation` from line 10

Remove definition: `retrieve_conversation`

(F811)